hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/impl/MetricsRecordImpl.java

```diff
@@ -27,7 +27,7 @@
 import org.apache.hadoop.metrics2.MetricsTag;
 import static org.apache.hadoop.metrics2.util.Contracts.*;
 
-class MetricsRecordImpl extends AbstractMetricsRecord {
+public class MetricsRecordImpl extends AbstractMetricsRecord {
```

Comment:
Could you add `@VisibleForTesting`?

Reply:
Thanks for your help reviewing the code, I will modify the code.